File: DatingApp-SPA/src/app/_models/UserDTO.ts
import { PhotoDTO } from './PhotoDTO';

export class UserDTO {
 id: number;
 fullName: string;
 username: string;
 password: string;

 knownAs?: string;

 genderId?: number;
 genderName: string;

 cityId?: number;
 cityName: string;
 // city?: any;

 countryId?: number;
 countryName: string;
 // country?: any;

 dateOfBirth?: Date;
 lastActive?: Date;
 createdOn?: Date;
 age?: number;
 fullAge?: string;
 photoUrl?: string;

 introduction: string;
 lookingFor: string;
 interests: string;

 photos?: PhotoDTO[];

 public UserDTO() { }
}

File: DatingApp.API/DatingApp.API/ModelsDTO/UserDTO.cs
using DatingApp.API.Models;
using System;
using System.Collections.Generic;

namespace DatingApp.API.ModelsDTO
{
    public class UserDTO
    {
        public int Id { get; set; }

        public string FullName { get; set; }

        public string Username { get; set; }

        public string KnownAs { get; set; }

        public int? GenderId { get; set; }
        public string GenderName { get; set; }
        //public Gender Gender { get; set; }

        public int? CityId { get; set; }
        public string CityName { get; set; }
        //public City City { get; set; }

        public int? CountryId { get; set; }
        public string CountryName { get; set; }
        //public Country Country { get; set; }


        public DateTime? DateOfBirth { get; set; }
        public DateTime? LastActive { get; set; }
        public DateTime? CreatedOn { get; set; }

        public string Introduction { get; set; }
        public string LookingFor { get; set; }
        public string Interests { get; set; }


        public ICollection<PhotoDTO> Photos { get; set; }

        #region Extra
        public int Age { get; set; }
        public string FullAge { get; set; }
        public string PhotoUrl { get; set; }
        #endregion
    }
}
